internal/dag/dag.go

```diff
@@ -349,17 +375,24 @@ func (d *DAG) recompute() *dag {
 				}
 			}
 			vhost(host, 80).routes[r.path] = r
+			continue
 		}
-	}
 
-	_d := new(dag)
-	for _, vh := range _vhosts {
-		_d.roots = append(_d.roots, vh)
-	}
-	for _, svh := range _svhosts {
-		_d.roots = append(_d.roots, svh)
+		// otherwise, if the route is delegating to another ingressroute, find it and process it.
```

Comment:
Could there be an error condition that a route has a delegate and a service defined? I think @rosskukulinski came up with this the other day. Given that, we should throw it out as an error.

Reply:
Per slack discussion, we'll handle this in status pr to 1. check for this status and 2. set it on the IngressRoute object